sipXconfig/neoconf/src/org/sipfoundry/sipxconfig/dialplan/config/ForwardingRules.java

```diff
@@ -90,7 +91,12 @@ public void write(Writer writer) throws IOException {
         context.put("proxyAddress", m_addressManager.getSingleAddress(ProxyManager.TCP_ADDRESS, getLocation()));
         context.put("statusAddress", m_addressManager.getSingleAddress(Mwi.SIP_TCP, getLocation()));
         context.put("regEventAddress", m_addressManager.getSingleAddress(Registrar.EVENT_ADDRESS, getLocation()));
-        context.put("regAddress", m_addressManager.getSingleAddress(Registrar.TCP_ADDRESS, getLocation()));
+        // Use Local registrar, if not available use global rr SRV record
+        if(m_featureManager.isFeatureEnabled(Registrar.FEATURE, getLocation())) {
+            context.put("regAddress", m_addressManager.getSingleAddress(Registrar.TCP_ADDRESS, getLocation()));
```

Comment:
what would happen if the local registrar currently not working? proxy won't try sending messages to a different one as it gets only the local registrar from DNS lookup, have you tried testing this scenario? Thanks!